docs/source/features/communication_overlap.rst

```diff
@@ -0,0 +1,61 @@
+Communication Overlap
+====================
+
+Data-parallel communication overlap
+-----------------------------------
+
+NeMo supports the overlap of the data-parallel (DP) communications with the computations in LLM training.
+NeMo features Distributed Optimizer that distributes optimizer states and the high precision master parameters across GPUs and this introduces two types of data-parallel communications; reduce-scatter of gradients and all-gather of updated parameters.
+The DP communication is chunked by the granularity of a Transformer layer and overlap each communication chunk with computation.
+This overlap method exposes only one DP communication chunk ensuring efficient large-scale LLM training.
+When training with pipeline-parallelism, the granularity of DP communication becomes the Transformer layers per virtual pipeline stage.
+
+DP gradient reduce-scatter and parameter all-gather overlaps are enabled when setting ``overlap_grad_sync=true`` and ``overlap_param_sync=true``, respectively.
+The precision of the gradient reduce-scatter is set by ``grad_sync_dtype`` and reduction in bf16 ensures improved performance at large scale training compared to the default precision of fp32.
+When training in fp8 computing precision (with ``fp8=true``), setting ``fp8_params=true`` conducts the parameter all-gather in fp8, reducing the all-gather overhead by half.
```

Comment:
To enable DP gradient reduce-scatter and parameter all-gather overlaps, you can set the following parameters:

- Overlap gradient synchronization: Set overlap_grad_sync=true. This ensures that gradient reduce-scatter overlaps with computation.

- Overlap parameter synchronization: Set overlap_param_sync=true. This allows parameter all-gather to overlap with computation.

Additionally, consider the following precision settings:

- Gradient reduce-scatter precision: Use grad_sync_dtype to set the precision. Choosing reduction=bf16 (bfloat16) can improve performance in large-scale training compared to the default precision of fp32.

- Parameter all-gather precision (when using fp8 computing): If you’re training with fp8=true, setting fp8_params=true conducts the parameter all-gather in fp8, reducing the all-gather overhead by half.